addons/partner_autocomplete/models/res_partner_autocomplete_sync.py

```diff
@@ -31,6 +32,9 @@ def start_sync(self):
 
             to_sync_item.write({'synched': True})
 
+        remaining_count = 0 if len(to_sync_items) < batch_size else self.search_count([('synched', '=', False)])
+        return len(to_sync_items), remaining_count
```

Comment:
```suggestion
        done = len(to_sync_items)
        remaining = 0 if done < batch_size else self.search_count([('synched', '=', False)])
        return (done, remaining)
```

Reply:
The more I read about those tuples the more I wonder if we shouldn't `return {'done': ..., 'remaining': ...}` instead. The code for testing that the server action indeed returned a progress would be safer too:
```diff
-if not result or not isinstance(result, Sequence):
-    return (0, None)
+if not isinstance(result, Mapping) or 'done' not in result or 'remaining' not in result:
+    result = {'done': 0, 'remaining': None}
```

I would also like that `_notify_progress` be keyword-only:
```python
def _notify_progress(self, *, done, remaining)
```